CMakeLists.txt

```diff
@@ -275,9 +275,13 @@ if (SINGULARITY_USE_HDF5)
   # hdf5, for data interopability
   # type: external
   enable_language(C)
+  set(_hdf5_comp "C;HL")
+  if(SINGULARITY_USE_FORTRAN)
+    set(_hdf5_comp "${_hdf5_comp};Fortran")
+  endif()
```

Comment:
Is this needed? I don't think the fortran HDF5 functionality is ever required.

Reply:
Perhaps it would be ok, but it seems that hdf5 through cmake is very fragile and subtle differences cause linking issues in host codes. When hdf5 is less fragile, perhaps we can get away with using this machinery as intended, but until then, I'm hesitant to allow this line to be different from the host codes.

Reply:
This actually is looking maybe a bit complicated: when building for EAP, there are some undefined symbols related to `libhdf5_fortran`. It's not clear to me if this is an issue with EAP build system. SAP is using hand-spun makefiles, so this may also be an issue with CMake handling libs. 

I know @dholladay00 had to do something with the `singularity-eos` hdf5 libs to build with EAP correctly, my intinct was it was the Fortran libs. My thinking here was to do what we're doing with MPI/HDF5, though there are differences that I know think have some deeper issue.

If there isn't objection, can we bring this into `main` for now?

Reply:
> When hdf5 is less fragile, perhaps we can get away with using this machinery as intended, but until then, I'm hesitant to allow this line to be different from the host codes.

I'm actually replicating the EAP HDF5 components here, (i.e. `find_package(HDF5 COMPONENTS C Fortran HL)`)

Reply:
Ok---if that's what it takes for the host codes, that's what it takes for the host codes.